app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -367,12 +371,83 @@ public boolean onOptionsItemSelected(final MenuItem item) {
             }
         } else if (item.getItemId() == R.id.menu_item_rename_playlist) {
             createRenameDialog();
-        } else {
-            return super.onOptionsItemSelected(item);
+        } else if (item.getItemId() == R.id.menu_item_remove_duplicates) {
+            if (!isRemovingDuplicateStreams) {
+                new AlertDialog.Builder(requireContext())
+                        .setMessage(R.string.remove_duplicates_popup_warning)
+                        .setTitle(R.string.remove_duplicates_popup_title)
+                        .setPositiveButton(R.string.yes,
+                                (DialogInterface d, int id) -> removeDuplicateStreams())
+                        .setNegativeButton(R.string.cancel,
+                                (DialogInterface d, int id) -> d.cancel())
+                        .create()
+                        .show();
+            } else {
+                return super.onOptionsItemSelected(item);
+            }
         }
         return true;
     }
 
+    public void removeDuplicateStreams() {
+        if (isRemovingDuplicateStreams) {
+            return;
+        }
+        isRemovingDuplicateStreams = true;
+        showLoading();
+
+        disposables.add(playlistManager.getPlaylistStreams(playlistId)
+                .subscribeOn(Schedulers.io())
+                .map((List<PlaylistStreamEntry> playlist) -> {
+                    // Playlist data
+                    final Iterator<PlaylistStreamEntry> playlistIter = playlist.iterator();
+
+                    // Remove duplicates, Functionality data
+                    final List<PlaylistStreamEntry> uniquePlaylistItems = new ArrayList<>();
+                    final Set<Long> uniquePlaylistItemStreamIds = new HashSet<>();
```

Comment:
Hash sets are unordered, right? Wouldn't that be an issue here, or am I mistaken?

Reply:
Hash sets are unordered, but it doesn't present a problem here because it's only being used to check if a stream has already been seen. The List preserves the order of the steams in the playlist.